contracts/pool/src/tests/liquidity.rs

```diff
@@ -698,3 +702,253 @@ fn withdraw_liqudity_below_min() {
+    let mut token2 = deploy_token_contract(&env, &admin2);
+    if token2.address < token1.address {
+        std::mem::swap(&mut token1, &mut token2);
+        std::mem::swap(&mut admin1, &mut admin2);
+    }
+    let user1 = Address::generate(&env);
+    let user2 = Address::generate(&env);
+    let user3 = Address::generate(&env);
+    let stake_manager = Address::generate(&env);
+    let stake_owner = Address::generate(&env);
+
+    let pool = deploy_liquidity_pool_contract(
+        &env,
+        None,
+        (&token1.address, &token2.address),
+        0i64,
+        None,
+        None,
+        None,
+        stake_manager,
+        stake_owner,
+    );
+
+    let share_token_address = pool.query_share_token_address();
+    let token_share = token_contract::Client::new(&env, &share_token_address);
+
+    token1.mint(&user1, &150);
+    token1.mint(&user2, &250);
+    token1.mint(&user3, &350);
+
+    token2.mint(&user1, &200);
+    token2.mint(&user2, &300);
+    token2.mint(&user3, &400);
+
+    // all users provide liquidity in a 3:4 ratio
+    pool.provide_liquidity(&user1, &Some(150), &Some(10), &Some(200), &Some(10), &None);
+    pool.provide_liquidity(&user2, &Some(150), &Some(50), &Some(200), &Some(50), &None);
+    pool.provide_liquidity(
+        &user3,
+        &Some(150),
+        &Some(100),
+        &Some(200),
+        &Some(100),
+        &None,
+    );
+
+    // user1 assertions
+    let lp_share_balance_user1 = token_share.balance(&user1);
+    let query_share_result_user1 = pool.query_share(&lp_share_balance_user1);
+    assert_eq!(
+        query_share_result_user1,
+        (
+            Asset {
+                address: token1.address.clone(),
+                amount: 149
+            },
+            Asset {
+                address: token2.address.clone(),
+                amount: 199
+            }
+        )
+    );
+
+    let pool_info_befor_withdrawal = pool.query_pool_info();
+    pool.withdraw_liquidity(&user1, &lp_share_balance_user1, &100i128, &100i128);
+    let pool_info_after_withdrawal = pool.query_pool_info();
+
+    assert!(pool_info_befor_withdrawal.asset_a.amount > pool_info_after_withdrawal.asset_a.amount);
+    assert!(pool_info_befor_withdrawal.asset_b.amount > pool_info_after_withdrawal.asset_b.amount);
+    assert!(
+        pool_info_befor_withdrawal.asset_lp_share.amount
+            > pool_info_after_withdrawal.asset_lp_share.amount
+    );
+
+    let lp_share_balance_after_withdraw_user1: i128 = token_share.balance(&user1);
+    assert_eq!(lp_share_balance_after_withdraw_user1, 0);
+
+    let query_share_result_user1 = pool.query_share(&lp_share_balance_after_withdraw_user1);
+    assert_eq!(
+        query_share_result_user1,
+        (
+            Asset {
+                address: token1.address.clone(),
+                amount: 0
+            },
+            Asset {
+                address: token2.address.clone(),
+                amount: 0
+            }
+        )
+    );
```

Comment:
Make multiple users bond at the same time, there's no point in bonding/unbonding each user separately.